### ISSUE TITLE
Fault tolerent job default global learning rate is not set

### ISSUE THREAD

**Issue by typhoonzero** [PaddlePaddle/Paddle#5005]
Running fault tolerent job with pserver written in go currently go to this [line](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/trainer/NewRemoteParameterUpdater.cpp#L109)

This is because the default learning rate for v2 API is always "poly": https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/trainer_config_helpers/optimizers.py#L362 , and the pserver does not support it yet. should implement "poly" method.

**Comment by helinwang** [comment]
Even though we are using constant learning rate policy, the model still does not converge, is it because the constant learning rate is not using the value from `trainerConfig_.learning_rate()` as the learning rate? And I have not see anywhere learning rate is set in that line, so I am curious what is the default value?

**Comment by typhoonzero** [comment]
@helinwang V2 API `optimizer` will call https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/trainer_config_helpers/optimizers.py#L362 , the learning rate policy is "poly", and this configuration will be convert to protobuf format `TrainerConfig.proto`.

**Comment by typhoonzero** [comment]
I did some detailed test today, the result is:

1. Start a fault tolerant mnist training job using [train_mnist_ft.py](https://github.com/PaddlePaddle/cloud/blob/develop/doc/autoscale_example/train_mnist_ft.py) with 5 trainers and 2 pservers. The job is running fine and can converge.
1. Scale down the job by decreasing 2 trainer pod(`kubectl create -f autoscale_load`), the job continues fine.
1. Scale up the job by adding 2 trainer pod(`kubectl delete -f autoscale_load`). The newly created trainer does not converge but the old trainers does:

Newly scaled trainer pod log:
```
label selector: paddle-job-master=example, desired: 1
label selector: paddle-job=example, desired: 2
Starting training job:  /tmp, num_gradient_servers: 2, trainer_id:  1, version:
I1024 12:07:55.353288    32 Util.cpp:166] commandline:  --num_gradient_servers=2 --ports_num_for_sparse=1 --use_gpu=0 --trainer_id=1 --trainer_count=1 --num_passes=1 --ports_num=1 --port=7164
[INFO 2017-10-24 12:07:55,364 layers.py:2556] output for __conv_pool_0___conv: c = 20, h = 24, w = 24, size = 11520
[INFO 2017-10-24 12:07:55,405 layers.py:2684] output for __conv_pool_0___pool: c = 20, h = 12, w = 12, size = 2880
[INFO 2017-10-24 12:07:55,411 layers.py:2556] output for __conv_pool_1___conv: c = 50, h = 8, w = 8, size = 3200
[INFO 2017-10-24 12:07:55,417 layers.py:2684] output for __conv_pool_1___pool: c = 50, h = 4, w = 4, size = 800
I1024 12:07:55.534462    32 GradientMachine.cpp:94] Initing parameters..
I1024 12:07:55.543933    32 GradientMachine.cpp:101] Init parameters done.
run test here
run test here
run test here
run test here
run test here
run test here
run test here
run test here
run test here
Pass 9, Batch 0, Cost 2.980498, {'classification_error_evaluator': 0.9140625}
run test here
Pass 10, Batch 0, Cost 2.312326, {'classification_error_evaluator': 0.8984375}
run test here
Pass 11, Batch 0, Cost 2.306100, {'classification_error_evaluator': 0.8984375}
```

Old trainer pod log:

```
label selector: paddle-job-master=example, desired: 1
label selector: paddle-job=example, desired: 2
Starting training job:  /tmp, num_gradient_servers: 2, trainer_id:  0, version:
I1024 11:38:16.641094    36 Util.cpp:166] commandline:  --num_gradient_servers=2 --ports_num_for_sparse=1 --use_gpu=0 --trainer_id=0 --trainer_count=1 --num_passes=1 --ports_num=1 --port=7164
[INFO 2017-10-24 11:38:16,683 layers.py:2556] output for __conv_pool_0___conv: c = 20, h = 24, w = 24, size = 11520
[INFO 2017-10-24 11:38:16,686 layers.py:2684] output for __conv_pool_0___pool: c = 20, h = 12, w = 12, size = 2880
[INFO 2017-10-24 11:38:16,692 layers.py:2556] output for __conv_pool_1___conv: c = 50, h = 8, w = 8, size = 3200
[INFO 2017-10-24 11:38:16,734 layers.py:2684] output for __conv_pool_1___pool: c = 50, h = 4, w = 4, size = 800
I1024 11:38:16.764544    36 GradientMachine.cpp:94] Initing parameters..
I1024 11:38:16.767114    36 GradientMachine.cpp:101] Init parameters done.
Pass 0, Batch 0, Cost 3.109155, {'classification_error_evaluator': 0.8984375}
run test here
Pass 1, Batch 0, Cost 0.870059, {'classification_error_evaluator': 0.25}
run test here
Pass 2, Batch 0, Cost 0.689995, {'classification_error_evaluator': 0.2109375}
run test here
Pass 3, Batch 0, Cost 0.526903, {'classification_error_evaluator': 0.171875}
run test here
Pass 4, Batch 0, Cost 0.287425, {'classification_error_evaluator': 0.0859375}
run test here
Pass 5, Batch 0, Cost 0.402872, {'classification_error_evaluator': 0.140625}
run test here
Pass 6, Batch 0, Cost 0.426507, {'classification_error_evaluator': 0.0703125}
Pass 6, Batch 100, Cost 0.272435, {'classification_error_evaluator': 0.0625}
run test here
Pass 7, Batch 0, Cost 0.048170, {'classification_error_evaluator': 0.015625}
Pass 7, Batch 100, Cost 0.084337, {'classification_error_evaluator': 0.0390625}
run test here
Pass 8, Batch 0, Cost 0.050831, {'classification_error_evaluator': 0.03125}
Pass 8, Batch 100, Cost 0.058482, {'classification_error_evaluator': 0.015625}
run test here
Pass 9, Batch 0, Cost 0.045268, {'classification_error_evaluator': 0.015625}
run test here
Pass 10, Batch 0, Cost 0.013293, {'classification_error_evaluator': 0.0}
run test here
Pass 11, Batch 0, Cost 0.127966, {'classification_error_evaluator': 0.046875}
```

The scale-up operation is at "pass 9", and it seems that old trainers is still affected by the scale operation.

**Comment by helinwang** [comment]
@typhoonzero thanks, I am looking into this, one question: from the log:
```
commandline:  --num_gradient_servers=2 --ports_num_for_sparse=1 --use_gpu=0 --trainer_id=1 --trainer_count=1 --num_passes=1 --ports_num=1 --port=7164
```
Is the new pserver being started (or the old pserver?)

**Comment by typhoonzero** [comment]
Sorry，in the above test case I've changed `is_local=True`. When training with remote pservers, the result is just like what you said.

**Comment by helinwang** [comment]
No worries!
